src/game_battler.cpp

```diff
@@ -341,11 +342,15 @@ bool Game_Battler::UseSkill(int skill_id, const Game_Battler* source) {
 					AddState(lcf::Data::states[i].ID, false);
 				}
 				else {
+					if (i == 0 && IsDead()) {
+						revived = 1;
+					}
```

Comment:
I've confirmed this bug exists in master. I haven't tested your fix yet. Will do so after other comments addressed.